alchemy/src/destroy.ts

```diff
@@ -146,7 +152,21 @@ export async function destroy<Type extends string>(
       await destroy(nestedScope, options);
     }
 
-    await scope.delete(instance[ResourceID]);
+    if (options?.replace == null) {
+      if (nestedScope) {
+        await destroy(nestedScope, options);
+      }
+      await scope.deleteResource(instance[ResourceID]);
```

Comment:
Not sure what null means and it doesn't seem to be represented in the types

Reply:
see other comment:
https://github.com/sam-goodwin/alchemy/pull/417#discussion_r2160711769